bril-ts/builder.ts

```diff
@@ -124,6 +124,32 @@ export class Builder {
     this.curFunction.instrs.push(instr);
   }
 
+  /**
+   * Checks whether the last emitted instruction in the current function is the specified op code.
+   * Useful for checking for terminating instructions.
+   */
+  isLastEmittedEffectOp(op: bril.EffectOpCode): boolean {
```

Comment:
Seems like useful behavior! But the name is a little odd, in two ways:

* `Emitted`: Maybe it would be a little more straightforward to just say it's the last op currently in the function? It seems less material that it's the last thing we emitted.
* `Effect`: I think this would look for both effect and value ops, right?

Another alternative, FWIW, would be to just get the last (non-label) instruction. So the return type would be `bril.Instruction?`, and callers could get this behavior by doing stuff like `lastInstr()?.op === "add"`.

Reply:
Sure, works for me.

I'll just note that I don't think the the `?` sugar works on return types? https://github.com/microsoft/TypeScript/issues/51332 Maybe I missed something obvious.

Instead I'm using `undefined` which the rest of the code base seems to do.